[Yap Xuan Xuan] iP

yapxuanxuan opened the pull request:
# Bobby
> Too much on your mind and want a useful chatbot to help you keep track of your tasks, deadlines and events? 
Fret not, Bobby is here! 😍 
Bobby is the child of Duke :) Bobby helps to keep track of the things you need to do so that you can free your mind for other things.  It is

- text-based
- easy to use

What are you waiting for? Download Bobby now [here](https://github.com/yapxuanxuan/ip) under releases.

Simply start by:

1. Downloading Bobby and double-clicking on it
2. Start adding tasks
3. Let it manage your tasks for you

best of all, it is 💥 **FREE**! 💥 

Now that you have Bobby, you will _not_ forget about your tasks anymore.  😁 

If you are a Java programmer, here is the sample code for how Bobby prints the introduction to the user and the `intro` method:

```java
public void intro() {
    System.out.print("Hello there! I'm Bobby, your handy assistant :>");
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yapxuanxuan committed:
level-0

yapxuanxuan committed:
level-0

yapxuanxuan committed:
level-0 final

yapxuanxuan committed:
level 1

yapxuanxuan committed:
level-1

yapxuanxuan committed:
OOP and abstraction level 2

yapxuanxuan committed:
level 3

yapxuanxuan committed:
try

yapxuanxuan committed:
try

yapxuanxuan committed:
level 4

yapxuanxuan committed:
level 5 test

yapxuanxuan committed:
level 5

yapxuanxuan committed:
level 6

yapxuanxuan committed:
Completed Automated Testing
can add more edge cases for bug fixing in the future

yapxuanxuan committed:
Added functions to allow saving in the hard disk and loading from it.

yapxuanxuan committed:
Stored deadline dates as java LocalDateTime

yapxuanxuan committed:
Merge branch 'Level-8'

yapxuanxuan committed:
Abstracted more classes, Parser, TaskList, UI and Storage

yapxuanxuan committed:
Merge branch 'A-MoreOOP'

yapxuanxuan committed:
Organized the classes into package duke and duke.task

yapxuanxuan committed:
Merge branch 'A-Packages'

yapxuanxuan committed:
Fixed bug for level-7 saving file in data.
Added code to check for whether the file exists and to create it if it doesn't exist.

yapxuanxuan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yapxuanxuan committed:
Added JUnit tests for all methods in Parser.java class.
Other classes mostly have void methods, not sure whether possible and how to test.

yapxuanxuan committed:
Merge branch 'A-JUnit'

yapxuanxuan committed:
Changed the packages to seedu.duke instead of just duke and created the duke.jar file.

yapxuanxuan committed:
Merge branch 'A-Jar'

yapxuanxuan committed:
Added Java Documentation accordingly.

yapxuanxuan committed:
Adjusted coded to comply with coding standards.

yapxuanxuan committed:
Added find function to duke, and fixed bug of deleting, marking or unmarking tasks that are not in the list.
Dealing with task numbers that are invalid.

yapxuanxuan committed:
Merge branch 'branch-A-JavaDoc'

yapxuanxuan committed:
Merge branch 'branch-A-CodingStandard'

yapxuanxuan committed:
Merge branch 'branch-Level-9'

ys112 commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,99 @@
+ * @version 0.1
+ */
+public class Duke {
+    private static Storage store;
```
Could a noun be used as a variable name here instead of "store"? (For code readability)

ys112 commented on src/main/java/seedu/duke/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+     * @param isNew boolean that indicates whether the task is an existing one loaded from
+     *              the file or a newly added one.
+     */
+    public void addItem(Task task, boolean isNew) { //to append items to tasks
```
Perhaps consider changing the method name to "addTask" for a better representation of the functions?

ys112 commented on src/main/java/seedu/duke/Storage.java:
```diff
@@ -0,0 +1,179 @@
+     *
+     * @param task the Task object whose information is to be stored.
+     */
+    public static void add(Task task) { //to append items to taskList
```
Would it be better if the method name is more detailed to reflect the function of adding task to file?
One possible suggestion is "addTaskToFile"

ys112 reviewed (commented):
LGTM. Well documented codes. I found some names that could be improved for further readability and understanding of the methods.

Murugan-Maniish commented on src/main/java/seedu/duke/UI.java:
```diff
@@ -0,0 +1,72 @@
+ * @version 0.1
+ */
+public class UI {
+    private Storage store;
+    private Scanner scanner;
+    public UI(Storage store) {
+        this.scanner = new Scanner(System.in);
+        this.store = store;
+    }
+    private static final String LINE =
+                "      _________________________________________________________________________\n";
+    private String logo = "     _______       ______     _______    _______    ___  ___\n"
+            + "    |   _  \"\\     /    \" \\   |   _  \"\\  |   _  \"\\  |\"  \\/\"  |\n"
+            + "    (. |_)  :)   // ____  \\  (. |_)  :) (. |_)  :)  \\   \\  /\n"
+            + "    |:     \\/   /  /    ) :) |:     \\/  |:     \\/    \\\\  \\/\n"
+            + "    (|  _  \\\\  (: (____/ //  (|  _  \\\\  (|  _  \\\\    /   /\n"
+            + "    |: |_)  :)  \\        /   |: |_)  :) |: |_)  :)  /   /\n"
+            + "    (_______/    \\\"_____/    (_______/  (_______/  |___/";
```
Might consider ordering class variables first, instance variables second followed by constructer before methods.

Murugan-Maniish commented on src/main/java/seedu/duke/task/Deadline.java:
```diff
@@ -0,0 +1,88 @@
+ * @version 0.1
+ */
+public class Deadline extends Task{
+    private LocalDateTime deadline;
+    public Deadline(String task, LocalDateTime deadline) {
+        super(task);
+        this.deadline = deadline;
+    }
```
Can consider adding a blank line between variable and constructer

Murugan-Maniish commented on src/main/java/seedu/duke/task/Event.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @Override
+    public void printTaskDesc(int num, boolean isLast){
```
A space before "{" would be ideal

Murugan-Maniish commented on src/main/java/seedu/duke/task/Task.java:
```diff
@@ -0,0 +1,101 @@
+    public void printTaskDesc(int num, boolean isLast){
+        if (!isLast) {
+            if (num == 1) {
+                System.out.print("      _________________________________________________________________________\n");
```
Should this be extracted out?

Murugan-Maniish commented on src/test/java/seedu/duke/ParserTest.java:
```diff
@@ -0,0 +1,94 @@
+            assertEquals(expected, Parser.parseEvent("event"));
+            fail();
+        } catch (DukeException e) {
+            assertEquals("      ______________________________________________________________________________________\n      Oops! Please state the description of your event and provide the start and end timing."
```
This line might be exceed the 120 char limit

Murugan-Maniish reviewed (commented):
LGTM! Some minor additions like extracting out the line could improve coding standards.

yapxuanxuan committed:
Created GUI for Bobby chatbot

yapxuanxuan committed:
Merge tag 'Level-10'

yapxuanxuan committed:
Used Java assertions to define assumptions about the program state.

Added assertions in Parser class to define the assumption that empty
inputs will not be passed to the functions in Parser class to be parsed.

Added assertions in TaskList class to define the assumption that the
task number has to be a valid array index, more than or equals to 0.

Added assertions in Task, Event and Deadline classes to define the
assumption that the task number has to be more than or equals to 1.

yapxuanxuan committed:
Merge branch 'A-Assertion'

yapxuanxuan committed:
Improved code quality.

Improved readability by deleting dead code and redundant variables.

Made the happy path prominent by using guard clauses at the top of the
function and throwing an exception if it enters the guard clause.

Minimized code duplication and edited my comments to be in the format of
writing to a reader.

yapxuanxuan committed:
Merge branch 'A-CodeQuality'

yapxuanxuan committed:
Added an extension C-Update.

With this new extension, users can edit their task details by
running a command in this format:
update <taskNum> /<desc/from/to/by> <newInfo>

Added methods in Parser class to parse this new command, as well as
methods in TaskList, Storage, Task, Deadline and Event classes to
support this new extension.

If the user tries to update a non-existent attribute for a task
(e.g updating a /by attribute for a todo task), a DukeException
will be thrown and printed in the DialogBox.

yapxuanxuan committed:
Merge branch 'C-Update'

yapxuanxuan committed:
Improving code quality by reducing code duplication.

Removed changeMarking method in Storage as it functions the same
way as updateFile method in Storage.

yapxuanxuan committed:
Merge pull request #2 from yapxuanxuan/C-Update

C-update

yapxuanxuan committed:
A-BetterGui Modified fxml files to improve GUI of Bobby.

Added background fill colours for user and Bobby's dialog box.

Added anchors to ensure that the window can be resized with the content
being resized nicely as well.

Clipped the image view with Circle shape so that the the user and
bobby's image looks more aesthetically pleasing.

Also modified the code logic to ensure that the data file to store the
task information is properly created if it does not exist when first
initialised.

yapxuanxuan committed:
Merge branch 'C-Update'

yapxuanxuan committed:
MMgit -c diff.mnemonicprefix=false -c core.quotepath=false --no-optional-locks push -v --tags origin master:master
Pushing to https://github.com/yapxuanxuan/ip.git
To https://github.com/yapxuanxuan/ip.git
 = [up to date]      A-Assertions -> A-Assertions
 = [up to date]      A-BetterGui -> A-BetterGui
 = [up to date]      A-CodeQuality -> A-CodeQuality
 = [up to date]      A-CodingStandard -> A-CodingStandard
 = [up to date]      A-FullCommitMessage -> A-FullCommitMessage
 = [up to date]      A-Gradle -> A-Gradle
 = [up to date]      A-JUnit -> A-JUnit
 = [up to date]      A-Jar -> A-Jar
 = [up to date]      A-JavaDoc -> A-JavaDoc
 = [up to date]      A-MoreOOP -> A-MoreOOP
 = [up to date]      A-Packages -> A-Packages
 = [up to date]      A-TextUiTesting -> A-TextUiTesting
 = [up to date]      BCD-Extension -> BCD-Extension
 = [up to date]      Level-10 -> Level-10
 = [up to date]      Level-7 -> Level-7
 = [up to date]      Level-8 -> Level-8
 = [up to date]      Level-9 -> Level-9
 = [up to date]      level-0 -> level-0
 = [up to date]      level-1 -> level-1
 = [up to date]      level-2 -> level-2
 = [up to date]      level-3 -> level-3
 = [up to date]      level-4 -> level-4
 = [up to date]      level-5 -> level-5
 = [up to date]      level-6 -> level-6
 ! [rejected]        master -> master (fetch first)
error: failed to push some refs to 'https://github.com/yapxuanxuan/ip.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.erge branch 'master' of https://github.com/yapxuanxuan/ip

yapxuanxuan committed:
Improve code quality

Deleted duplicate code.

yapxuanxuan committed:
Merge branch 'master' of https://github.com/yapxuanxuan/ip

yapxuanxuan committed:
Gave Bobby more personality.

Edited Bobby's responses to sound more friendly.

Renamed all Duke names to Bobby respectively.

yapxuanxuan committed:
Merge branch 'A-Personality'

yapxuanxuan committed:
Tackled more error handling.

Handled the error cases for the extension function, C-Update.

Handled the error where the user states a task number bigger than the
size of the task list. (invalid task number) Also handled the error
when the user does not state the attribute to update and indicate with
"/" character. Handled these error cases by throwing BobbyException.

yapxuanxuan committed:
Updated README.md

Updated the readme to introduce Bobby as well as how to use it.

Also added references in my code to give credit to the original authors.

yapxuanxuan committed:
Rename Duke to Bobby

Edited to ensure that all instances of Duke name is replaced with
Bobby instead, in comments, documentation and also the file name of
the text file used to store the information of the tasks on the
hard disk.

yapxuanxuan committed:
Added Ui.png

Added the screenshot of the product to the docs folder.

yapxuanxuan committed:
Added detailed User Guide.

Added guide in README under docs to help new users understand how to
use Bobby.

Also modified the time for program to exit after bye command to be 1s
instead of 1.25s.

yapxuanxuan committed:
Merge commit '8aada6f327ac0ec206582d9a0a11f4f113222488'

yapxuanxuan committed:
Added more JavaDoc documentation.

Amended the JUnit tests in ParserTest.java to be updated with the
current code as it was previously testing for the previous version
of code.
